yupiik-tools-maven-plugin/src/main/java/io/yupiik/maven/mojo/GenerateBlogReleaseFromGithubMojo.java

```diff
@@ -0,0 +1,379 @@
+
+/**
+ * Fetch releases from github repo to generate a blog post.
+ * The search is between 2 dates.
+ */
+@Mojo(name = "generate-blog-post-releases", threadSafe = true)
+public class GenerateBlogReleaseFromGithubMojo extends AbstractMojo {
+    @Parameter(property = "yupiik.generate-blog-post-releases.forceHttpV1", defaultValue = "true")
+    protected boolean forceHttpV1;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.useMavenCredentials", defaultValue = "true")
+    protected boolean useMavenCredentials;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.githubServerId", defaultValue = "github.com")
+    protected String githubServerId;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.githubRepository")
+    protected List<String> githubRepositories;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.githubRepository", defaultValue = "https://api.github.com/")
+    protected String githubBaseApi;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.threads", defaultValue = "16")
+    protected int threads;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.workdir", defaultValue = "${project.build.directory}/generate-blog-post-releases-workdir")
+    protected String workdir;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.from")
+    protected String from;
+
+    @Parameter(property = "yupiik.generate-blog-post-releases.to")
+    protected String to;
+
+    @Parameter(defaultValue = "${session}", readonly = true)
+    private MavenSession session;
+
+    @Component
+    private SettingsDecrypter settingsDecrypter;
+
+    private final Map<String, Optional<Server>> servers = new ConcurrentHashMap<>();
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        final var threadPool = Executors.newFixedThreadPool(threads, new ThreadFactory() {
+            private final AtomicInteger counter = new AtomicInteger();
+
+            @Override
+            public Thread newThread(final Runnable r) {
+                final var thread = new Thread(r, GenerateBlogReleaseFromGithubMojo.class.getSimpleName() + '-' + counter.incrementAndGet());
+                thread.setContextClassLoader(GenerateBlogReleaseFromGithubMojo.class.getClassLoader());
+                return thread;
+            }
+        });
+        final var httpClientBuilder = HttpClient.newBuilder();
+        if (forceHttpV1) {
+            httpClientBuilder.version(HttpClient.Version.HTTP_1_1);
+        }
+        final var httpClient = httpClientBuilder
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .executor(threadPool)
+                .build();
+        try (final Jsonb jsonb = JsonbBuilder.create(new JsonbConfig().setProperty("johnzon.skip-cdi", true))) {
+            final var workDir = Files.createDirectories(Paths.get(workdir));
+            Files.deleteIfExists(Path.of(workDir + "/releases-blog-post.adoc"));
```

Comment:
is it needed? if you open it without options it will overwrite it anyway at https://github.com/yupiik/tools-maven-plugin/pull/7/files#diff-7a9a6ae89337aab9648e9b123bfc69faa4c9f7f0fc9ee020b1856df90709e6e2R180 no?

Reply:
Yes because if you run it several time, it will append so we need to clean up at each run.

Reply:
Not really, it does cause you asked to append, if you dont it does not

Reply:
We append for each repository, so another option will be to write to the file at the end.

Reply:
change flow